hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/RatisPipelineUtils.java

```diff
@@ -51,6 +52,15 @@
   private static final Logger LOG =
       LoggerFactory.getLogger(RatisPipelineUtils.class);
 
+  // Set parallelism at 3, as now in Ratis we create 1 and 3 node pipelines.
+  private static int parallelismForPool =
+      (Runtime.getRuntime().availableProcessors() > 3) ? 3 :
+          Runtime.getRuntime().availableProcessors();
+
+  private static ForkJoinPool pool = new ForkJoinPool(parallelismForPool);
```

Comment:
We can use a simple executor as fork join pool is an executor for fork join tasks.

Reply:
Used ForkJoinPool can be used for normal tasks also. Refer NonFrok Join Clients.
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ForkJoinPool.html

Used this ForJoinPool here to take advantage of its WorkSteal Algorithm in allocating/claiming threads. I think this might be useful. More can be read from the documentation. 


And if no of available processors is less than 3, even setting to 3 will not have any advantage. This is the reason for setting this.

Reply:
We can use three threads as these threads would ultimately be blocked on network IO and will not be CPU intensive. Also we need to shutdown the fork join pool.

Reply:
Done